pennylane/drawer/draw.py

```diff
@@ -119,6 +119,9 @@ def circuit(a, w):
     [-0. -1. -0. -0.]
     [-0. -0. -1. -0.]
     [-0. -0. -0. -1.]]
+    >>> print(qml.draw(circuit3, show_matrices=False)())
+    0: ──U(M0)─╭U(M1)─┤
+    1: ────────╰U(M1)─┤  <𝓗(M0)>

```

Comment:
Are there any other example codeblocks outside of the `drawer` module that might be change by this PR?

Reply:
That took a while to get back to :D I went over the entire codebase and found a few instances of tapes being drawn that contain matrix parameters. A nice insight: More tapes used `show_matrices=True` explicitly than not, convincing me a bit more that this switch of default makes sense :)

Reply:
iirc the reason we originally turned this off by default was we had someone draw a circuit with a very big matrix, and it was causing issues 😅 Although I can't remember any other details.

Reply:
Would it be unreasonable to ask users to switch matrix printing off in such cases? :thinking:

Reply:
@josh146 just following up here. Do you think this would be a problem?

Reply:
> Would it be unreasonable to ask users to switch matrix printing off in such cases? 🤔

Depends on the behaviour I suppose. If a user calls `qml.draw(..)` without thinking on a circuit with a huge matrix, and the effect is that it could cause the python kernel to hang/crash etc, then that's a case where I would err on not having it be default behaviour

Reply:
Seems like we fall back to the smart numpy/python printing in this case (same for a list of lists object):
![Screenshot from 2023-05-16 09-08-11](https://github.com/PennyLaneAI/pennylane/assets/20772922/5dca91b3-2927-4874-90a7-24f05d6e80e4)

Reply:
💯 satisfies my concern then!